src/System.Text.Json/tests/JsonElementCloneTests.cs

```diff
@@ -0,0 +1,167 @@
+        }
+
+        [Fact]
+        public static void CloneInnerElementFromClonedElement()
+        {
+            JsonElement clone;
+
+            using (JsonDocument doc = JsonDocument.Parse("[[[]]]"))
+            {
+                JsonElement middle = doc.RootElement[0].Clone();
+                JsonElement inner = middle[0];
+                clone = inner.Clone();
+
+                Assert.Equal(inner.GetRawText(), clone.GetRawText());
+                Assert.NotSame(doc, clone.SniffDocument());
+                Assert.Same(middle.SniffDocument(), clone.SniffDocument());
+                Assert.Same(inner.SniffDocument(), clone.SniffDocument());
+            }
+
+            // After document Dispose
+            Assert.Equal("[]", clone.GetRawText());
+        }
+
+        [Fact]
+        public static void CloneAtInnerNumber()
+        {
+            CloneAtInner("1.21e9", JsonValueType.Number);
+        }
+
+        [Fact]
+        public static void CloneAtInnerString()
+        {
+            CloneAtInner("\"  this  string  has  \\u0039 spaces\"", JsonValueType.String);
+        }
+
+        [Fact]
+        public static void CloneAtInnerTrue()
+        {
+            CloneAtInner("true", JsonValueType.True);
+        }
+
+        [Fact]
+        public static void CloneAtInnerFalse()
+        {
+            CloneAtInner("false", JsonValueType.False);
+        }
+
+        [Fact]
+        public static void CloneAtInnerNull()
+        {
+            CloneAtInner("null", JsonValueType.Null);
+        }
+
+        [Fact]
+        public static void CloneAtInnerObject()
+        {
+            CloneAtInner(
+                @"{
+  ""this"":
+  [
+    {
+      ""object"": 0,
+
+
+
+
+      ""has"": [ ""whitespace"" ]
+    }
+  ]
+}",
+                JsonValueType.Object);
+        }
+
+        [Fact]
+        public static void CloneAtInnerArray()
+        {
+            CloneAtInner(
+                @"[
+{
+  ""this"":
+  [
+    {
+      ""object"": 0,
+
+
+
+
+      ""has"": [ ""whitespace"" ]
+    }
+  ]
+},
+
+5
+
+,
+
+
+
+false,
+
+
+
+null
+]",
+                JsonValueType.Array);
+        }
+
+        private static void CloneAtInner(string innerJson, JsonValueType valueType)
+        {
+            string json = $"{{ \"obj\": [ {{ \"not target\": true, \"target\": {innerJson} }}, 5 ] }}";
+
+            JsonElement clone;
+
+            using (JsonDocument doc = JsonDocument.Parse(json))
+            {
+                JsonElement target = doc.RootElement.GetProperty("obj")[0].GetProperty("target");
+                Assert.Equal(valueType, target.Type);
+                clone = target.Clone();
+            }
+
+            Assert.Equal(innerJson, clone.GetRawText());
+        }
+
+        private static JsonDocument SniffDocument(this JsonElement element)
+        {
+            return (JsonDocument)typeof(JsonElement).
+                GetField("_parent", BindingFlags.Instance|BindingFlags.NonPublic).
```

Comment:
Should we add a comment on the field to warn people when renaming/moving it to also update this test?

Reply:
Nah, the test'll just fail in a very obvious way.